HerPublicWebsite.BusinessLogic/Services/CsvFileCreator/CsvFileCreator.cs

```diff
@@ -29,7 +29,179 @@ public MemoryStream CreateFileData(IEnumerable<ReferralRequest> referralRequests
         }
     }
 
-    private class CsvRow
+    public MemoryStream CreateReferralRequestOverviewFileData(IEnumerable<ReferralRequest> referralRequests)
+    {
+        var rows = referralRequests.Select(rr => new CsvRowReferralCodes(rr));
+
+        var csvConfiguration = new CsvConfiguration(CultureInfo.InvariantCulture)
+        {
+            InjectionOptions = InjectionOptions.Strip
+        };
+
+        using var writeableMemoryStream = new MemoryStream();
+        using var streamWriter = new StreamWriter(writeableMemoryStream, Encoding.UTF8);
+        using var csvWriter = new CsvWriter(streamWriter, csvConfiguration);
+        {
+            csvWriter.WriteRecords(rows);
+            csvWriter.Flush();
+            return new MemoryStream(writeableMemoryStream.GetBuffer(), 0, (int)writeableMemoryStream.Length, false);
+        }
+    }
+
+    public MemoryStream CreateReferralRequestFollowUpData(IEnumerable<ReferralRequest> referralRequests)
+    {
+        var consortiumData = referralRequests.GroupBy(rr => LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[rr.CustodianCode].Consortium)
+            .Select(group => new CsvRowConsortiumFollowUpInformation(group));
+
+        var rows = referralRequests.GroupBy(rr => rr.CustodianCode)
+            .Select(group => new CsvRowLaDownloadInformation(
+                group, consortiumData.First(csd => csd.Consortium == (LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[group.Key].Consortium ?? "")))
```

Comment:
I feel like I've made this more complicated than it needs to be.

Possibly stems from an issue of adding the consortiums only as strings in the LocalAuthorityData table. Do you think this should be a separate table with some uuid reference/link between the LocalAuthorityData and consortium name?

Reply:
That looks much more clear. I've kept a ConsortiumStatistics class with the calculations done in the initialiser for making it a bit easier to read. But let me know if you think it should actually all be in this function.

Reply:
I've also given it a local test and it seems to be working as expected.